apps/sms/js/thread_ui.js

```diff
@@ -2049,6 +2049,12 @@ var ThreadUI = {
       case 'contextmenu':
         evt.preventDefault();
         evt.stopPropagation();
+
+        // Disable contextmenu while selecting non-editable message content
+        if (!this.threadMessages.classList.contains('editable-select-mode')) {
```

Comment:
I don't really like that you use the DOM to check the state.

I have 2 suggestions:

1. (my favorite) remove the "contextmenu" handler when in select mode, add it back when we exit select mode. We can replace `this.handleEvent.bind(this)` by `this` in https://github.com/mozilla-b2g/gaia/blob/1e2c128e1343668fe0ef7f8e48bf6cb60280dd19/apps/sms/js/thread_ui.js#L179-L181 which will make this very easy.

2. use a boolean state instead of the class. But if we can avoid having a state, I'd prefer this.

The only question I have is what happens when we longpress if we remove the "contextmenu" event; I guess the default "contextmenu" action will happen, but is it disabled by system when we're in selection mode?